src/ucp/tag/tag_send.c

```diff
@@ -250,17 +250,28 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_tag_send_nbx,
     ucs_trace_req("send_nbx buffer %p count %zu tag %"PRIx64" to %s",
                   buffer, count, tag, ucp_ep_peer_name(ep));
 
-    if (ucs_likely(!(param->op_attr_mask &
-                     (UCP_OP_ATTR_FIELD_DATATYPE | UCP_OP_ATTR_FLAG_NO_IMM_CMPL)))) {
+    switch(param->op_attr_mask & (UCP_OP_ATTR_FIELD_DATATYPE | UCP_OP_ATTR_FLAG_NO_IMM_CMPL)) {
```

Comment:
1. space after switch
2. long line: use local helper variable
3. pls use if/else instead of switch/case, fast path (no datatype) is first to check